fractal_server/app/models/workflow.py

```diff
@@ -161,7 +170,12 @@ async def insert_task(
             wt_meta = None
 
         # Create DB entry
-        wf_task = WorkflowTask(task_id=task_id, args=actual_args, meta=wt_meta)
+        wf_task = WorkflowTask(
+            workflow_id=self.id,
```

Comment:
This line (`workflow_id=self.id`) looks perfectly correct, but I'm confused: how could the previous version work? Where did we establish this foreign-key?

Reply:
The suspicion is that since it is `Optional` it is often null... can we verify it in a db?

Reply:
On a realistic DB (and from within psql):
```
# select id,workflow_id,task_id from workflowtask;
  id  | workflow_id | task_id 
------+-------------+---------
...
(678 rows)

# select id,workflow_id,task_id from workflowtask where workflow_id is null;
 id | workflow_id | task_id 
----+-------------+---------
(0 rows)

# select id,workflow_id,task_id from workflowtask where task_id is null;
 id | workflow_id | task_id 
----+-------------+---------
(0 rows)
```

Reply:
> This line (`workflow_id=self.id`) looks perfectly correct, but I'm confused: how could the previous version work? Where did we establish this foreign-key?

I'll answer myself: it was working because of
```python
class Workflow(_WorkflowBase, SQLModel, table=True):
    ...
    task_list: list[WorkflowTask] = Relationship(...)
    ...
    async def insert_task(...):
        ...
        self.task_list.insert(order, wf_task)
```
which also updates the (list of) relationship.